### PR TITLE
Update module-reference.md

### PR DESCRIPTION
**Pull request by MickL**
I tried to make the point about instantiating providers dynamically more clear, as I struggled with it.

## PR Checklist
Please check if your PR fulfills the following requirements:

- [X] The commit message follows our guidelines: https://github.com/nestjs/docs.nestjs.com/blob/master/CONTRIBUTING.md

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[X] Other... Please describe: Updated documentation
```

### CODE REVIEW

File: content/fundamentals/module-reference.md [hunk 1]
```diff
@@ -191,17 +191,32 @@ const catsRepository = await this.moduleRef.resolve(CatsRepository, contextId);
 
 #### Instantiating classes dynamically
 
-To dynamically instantiate a class that wasn't previously registered as a provider, use the module reference's `create()` method.
+To dynamically instantiate a class that wasn't previously registered as a provider but depends on other providers use the module reference's `create()` method.
```
Comment (jmcdo29) [inline content/fundamentals/module-reference.md:194]: Do these other providers need to be available in the current module's scope (i.e. Logger needs to be in the `providers` array)?

**Comment by johnbiundo** [comment]
I don't know the answer to @jmcdo29 's question; @kamilmysliwiec do you have a comment on that?

@MickL I'm not sure I understand this any better with your example.  Could you please elaborate in this discussion thread what the core issue is (i.e., your statement of the need that this feature is fulfilling, and how your example shows that)?  I'm struggling to connect the dots.  If you can do that in this discussion thread, maybe we can make some improvements to the actual doc.  Thanks!

**Comment by MickL** [comment]
The main issue was that I wanted to create a class dynamically e.g. `new Cat('Meowth');` but it should have the dependency injection so I can have existing services in there like the logger. The answer doing so was found in the documentation, but it was not clear to me. I wanted to alter it so others will find the informations doing so more easy.

File: content/fundamentals/module-reference.md (hunk 1 above)
Comment (kamilmysliwiec) [reply content/fundamentals/module-reference.md:194]: @jmcdo29 Yes

File: content/fundamentals/module-reference.md [hunk 2]
```diff
@@ -191,17 +191,32 @@ const catsRepository = await this.moduleRef.resolve(CatsRepository, contextId);
...
 @Injectable()
 export class CatsService implements OnModuleInit {
-  private catsFactory: CatsFactory;
+  private myCat: Cat;
   constructor(private moduleRef: ModuleRef) {}
 
   async onModuleInit() {
-    this.catsFactory = await this.moduleRef.create(CatsFactory);
+    this.myCat = await this.moduleRef.create(Cat);
+    this.myCat.init('Meowth');
```
Comment (kamilmysliwiec) [inline content/fundamentals/module-reference.md:219]: Can we revert this change? With the text update you proposed, everything should be clear now

**Comment by kamilmysliwiec** [comment]
Just replied @johnbiundo

File: content/fundamentals/module-reference.md (hunk 1 above)
Comment (jmcdo29) [reply content/fundamentals/module-reference.md:194]: I had a feeling that was the case. It should probably be explicitly mentioned if this is going to move forward.

**Comment by johnbiundo** [comment]
@jmcdo29 can you propose a final update to this?  I'm a little unclear after the back and forth where you guys think this should end up.  I can take a final pass on any style/grammar.

File: content/fundamentals/module-reference.md (hunk 2 above)
Comment (MickL) [reply content/fundamentals/module-reference.md:219]: Creating a factory was kind of confusing to me. I think it should be clear that when you can create a class, you can also create it using a factory. Or am I wrong?

**Comment by jmcdo29** [comment]
@johnbiundo I'll see if I have some time soon to propose some changes. This PR and #1129 kind of go hand in hand with how some of the parts work, so I'll try to help coordinate that

**Comment by MickL** [comment]
@jmcdo29 I appreciate that. For me it was very hard to find the right words but I hope you get what I wanted to archive with both PRs.

**Comment by johnbiundo** [comment]
@jmcdo29 Thanks.  I posted a suggestion on #1129 earlier as well.  No substantive changes, just style.

File: content/fundamentals/module-reference.md (hunk 2 above)
Comment (kamilmysliwiec) [reply content/fundamentals/module-reference.md:219]: `Factory` acts as a random (sample) name here. We can change it to, let's say, `CatsHelpers` or `CatsStorage`, or `CatsUtils`. I refer to the rest of the additional code (which seems unnecessary to me)

**Comment by adelin-b** [comment]
A suggestion for making it even more useful would be showing how to `this.moduleRef.create(Service)` and passing argument to the Service.constructor
